pkg/logbridge/logbridge.go

```diff
@@ -6,50 +6,80 @@ import (
 	"io"
 	"time"
 
+	"github.com/square/p2/Godeps/_workspace/src/github.com/rcrowley/go-metrics"
 	"github.com/square/p2/pkg/logging"
 )
 
-// Copy implements a buffered copy operation between dest and src.
+type LogBridge struct {
+	Reader        io.Reader
+	DurableWriter io.Writer
+	LossyWriter   io.Writer
+
+	logger        logging.Logger
+	metrics       MetricsRegistry
+	logLinesCount metrics.Counter
+	logBytes      metrics.Counter
+}
+
+type MetricsRegistry interface {
+	Register(metricName string, metric interface{}) error
+}
+
+func NewLogBridge(r io.Reader,
+	durableWriter io.Writer,
+	lossyWriter io.Writer,
+	logger logging.Logger,
+	metricsRegistry MetricsRegistry,
+	loglineMetricKeyName string,
```

Comment:
We should standardize on a way to name metrics, as the convention elsewhere in this repo is to name the keys whatever we please when the registry is passed to us (ie they are constants). Check consul_aggregator.go for an example.

I'm not married by any means to that approach, in fact I'd almost prefer a central location for metric names - lets dedicate a few minutes of time to figuring that out.